vscode/src/format.ts

```diff
@@ -37,10 +49,17 @@ class QSharpFormatProvider implements vscode.DocumentFormattingEditProvider {
       return [];
     }
 
-    const edits = lsEdits.map(
+    let edits = lsEdits.map(
       (edit) => new vscode.TextEdit(toVscodeRange(edit.range), edit.newText),
     );
 
+    if (range) {
+      edits = edits.filter(
+        (e) =>
+          range.start.isBefore(e.range.end) && range.end.isAfter(e.range.start),
```

Comment:
Does `range.intersection(e.range) !== undefined` do the trick here?

Also confirming: you want all the edits that overlap the requested range, not just the ones wholly contained in the range?

Reply:
Yes we want the edits that overlap.

The `range.intersection(e.range) !== undefined` has slightly different behavior that is not desirable. That one is equal to `range.start.isBeforeOrEqual(e.range.end) && range.end.isAfterOrEqual(e.range.start)`, causing edits to happen if they are adjacent to the highlighted text, not inside the highlighted text.

Reply:
Okay, fair enough!